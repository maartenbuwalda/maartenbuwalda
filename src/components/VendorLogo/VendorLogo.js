import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import images from '../../constants/images'

const getLogo = vendor => {
  switch (vendor) {
    case 'jest': return images.jest
    case 'react native':
    case 'react': return images.react
    case 'redux': return images.redux
    case 'ember': return images.ember
    case 'middleman': return images.middleman
    case 'storybook': return images.storybook
    default: return null
  }
}

const Figure = styled.figure`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 1rem 0 0;
  height: 3rem;
  width: 3rem;

  &:empty {
    display: none;
  }
`

const StyledLogo = styled.img`
  max-width: 3rem;
  max-height: 3rem;
  display: block;
`

const VendorLogo = ({ vendor }) => {
  const src = getLogo(vendor)
  return (
    <Figure>
      {src && <StyledLogo src={src} />}
    </Figure>
  )
}

VendorLogo.propTypes = {
  vendor: PropTypes.string,
}

export default VendorLogo
