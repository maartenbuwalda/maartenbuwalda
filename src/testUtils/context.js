const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
