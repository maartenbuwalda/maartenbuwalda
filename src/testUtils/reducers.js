export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
