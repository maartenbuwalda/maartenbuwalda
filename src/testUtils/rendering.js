import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
