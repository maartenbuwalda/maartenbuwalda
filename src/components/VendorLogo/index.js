import VendorLogo from './VendorLogo'

export default VendorLogo
