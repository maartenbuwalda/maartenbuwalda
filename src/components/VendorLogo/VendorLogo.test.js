import React from 'react'
import VendorLogo from './index'
import { componentRenders } from '../../testUtils/rendering'

jest.mock('../../constants/images', () => ({}))

const vendors = [ 'react', 'react native', 'jest', 'ember', 'redux', 'middleman', 'storybook' ]

describe('VendorLogo component', () => {
  describe('when doing a basic render', () => {
    const component = <VendorLogo />
    componentRenders(component)
  })

  vendors.forEach(vendor => {
    describe(`when passing ${vendor} as vendor`, () => {
      const component = <VendorLogo vendor={vendor} />
      componentRenders(component, vendor)
    })
  })
})
