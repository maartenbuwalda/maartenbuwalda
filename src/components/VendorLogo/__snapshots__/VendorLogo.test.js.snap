// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorLogo component when doing a basic render renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when fully mounting mounts correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 1rem 0 0;
  height: 3rem;
  width: 3rem;
}

.c0:empty {
  display: none;
}

<VendorLogo>
  <styled.figure>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 1rem 0 0;
  height: 3rem;
  width: 3rem;

  &:empty {
    display: none;
  }
",
            ],
          },
          "displayName": "styled.figure",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "figure",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <figure
        className="c0"
      />
    </StyledComponent>
  </styled.figure>
</VendorLogo>
`;

exports[`VendorLogo component when passing ember as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing jest as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing middleman as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing react as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing react native as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing redux as vendor renders correctly 1`] = `<styled.figure />`;

exports[`VendorLogo component when passing storybook as vendor renders correctly 1`] = `<styled.figure />`;
